lib/auth/permissions.go

```diff
@@ -173,11 +175,31 @@ func (a *authorizer) Authorize(ctx context.Context) (*Context, error) {
 	if ctx == nil {
 		return nil, trace.AccessDenied("missing authentication context")
 	}
-	userI := ctx.Value(ContextUser)
+	userI := ctx.Value(utils.ContextUser)
 	authContext, err := a.fromUser(ctx, userI)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	if authContext.Checker.PinSourceIP() {
+		var ip string
+		if md, ok := metadata.FromIncomingContext(ctx); ok {
```

Comment:
This can be pulled from `utils.ContextClientAddr` rather than from the `metadata` right ?